### DISCUSSION TITLE
Should the `client_secret_post` authentication method support empty/undefined client secrets ?

### DISCUSSION THREAD

**Discussion by NSeydoux** [panva/openid-client#430] [Q&A] [answered]
When building a Client instance for a given Issuer, if the client has an identifier but no secret (that may be the case in [Solid-OIDC](https://solid.github.io/solid-oidc/), an OIDC extension), the following error is thrown: `client_secret_post client authentication method requires a client_secret`.

My understanding of [the OAuth 2.0 spec](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1) is that not including the `client_secret` in the request body was acceptable, even if the parameter is marked as required, because it is possible to omit it for empty secrets. 

Is this understanding correct, in which case I'll open a proper issue, or am I misunderstanding the spec text ?

**Comment by panva** [comment, marked as answer]
You are describing the `none` [client auth method](https://github.com/panva/node-openid-client/blob/main/docs/README.md#client-authentication-methods) .
